docs/spec/api.md

```diff
@@ -673,8 +684,159 @@ The response will be in the following format:
     }
 
 For repositories with a large number of tags, this response may be quite
-large, so care should be taken by the client when parsing the response to
-reduce copying.
+large. If such a response is expected, one should use the pagination.
+
+#### Pagination
+
+Paginated tag results can be retrieved with by adding the appropriate
+pagination parameters. Starting a paginated flow may begin as follows:
+
+```
+GET /v2/<name>/tags/list?n=<integer>
+```
+
+The above specifies that a tags response should be returned, from the start of
+the result set, ordered lexically, limiting the number of results to `n`. The
+response to such a request would look as follows:
+
+```
+200 OK
+Content-Type: application/json
+
+{
+  "name": <name>,
+  "tags": [
+    <tag>,
+    ...
+  ]
+  "next": <url>?n=<n from the request>&last=<last tag value from previous response>
+}
+```
+
+> __TODO(stevvooe):__ Consider using a Header here, rather than a body parameter. A
```

Comment:
Any objections to leveraging the [RFC5988](https://tools.ietf.org/html/rfc5988) `Link` header? For example, with a domain added:

```
Link: <https://myregistry.foo/v2/_catalog?n=20&last=b>; rel="next"
```

This is super clean. The only issue is that it not yet a ratified standard.

@dmcgowan @wking @RichardScothern @endophage